backend/controller/SequenceController.ts

```diff
@@ -0,0 +1,382 @@
+ */
+export class SequenceController {
+    /**
+     * Creates a new sequence in the database with the given title and given user as owner.
+     * @param name the title of the sequence
+     * @param userId the userid of the owner
+     * @returns the created sequence
+     */
+    public static async createNewSequence(
+        name: string,
+        userId: number
+    ): Promise<ISequence> {
+        if (!this.isValidUser(userId)) {
+            throw new LoernwerkError(
+                'user not found',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        const dbSequence = new DBSequence();
+        let newCode = this.generateCode();
+        while ((await DBSequence.findBy({ code: newCode })).length > 0) {
+            newCode = this.generateCode();
+        }
+        dbSequence.code = newCode;
+        dbSequence.authorId = userId;
+        dbSequence.name = name;
+        dbSequence.slideCount = 0;
+        dbSequence.writeAccess = [];
+        dbSequence.readAccess = [];
+        await dbSequence.save();
+        return dbSequence;
+    }
+
+    /**
+     * Searches the database for a sequence with corresponding access code. Throws an error if no sequence could be found.
+     * @param code the code of the sequence
+     * @returns the sequence
+     */
+    public static async getSequenceByCode(code: string): Promise<ISequence> {
+        const dbSequence = await DBSequence.findOneBy({
+            code: code.toUpperCase(),
+        });
+        if (dbSequence === null) {
+            throw new LoernwerkError(
+                'no matching sequence',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        return dbSequence;
+    }
+
+    /**
+     * Searches the database for a sequence with corresponding access code, as well as its slides. Throws an error if no sequence could be found.
+     * @param code the code of the sequence
+     * @returns the sequence with slides
+     */
+    public static async getSequenceWithSlides(
+        code: string
+    ): Promise<ISequenceWithSlides> {
+        const sequenceWithoutSlide = await this.getSequenceByCode(code);
+        if (sequenceWithoutSlide === null) {
+            throw new LoernwerkError(
+                'No sequence found',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        const sequenceWithSlide = sequenceWithoutSlide as ISequenceWithSlides;
+        sequenceWithSlide.slides = await DBSlide.find({
+            where: { sequenceCode: code.toUpperCase() },
+        });
+        sequenceWithSlide.slideCount = sequenceWithSlide.slides.length;
+        return sequenceWithSlide;
+    }
+
+    /**
+     * Save the sequence in the database
+     * @param sequence the sequence
+     */
+    public static async saveSequence(
+        sequence: Partial<ISequenceWithSlides>
+    ): Promise<void> {
+        if (sequence.code == undefined) {
```

Comment:
===

Reply:
@ProgramPhoenix You unresolved this and did not change it. Was that on purpose?

Reply:
defenitly changed localy

Reply:
now its also updated in github for me